build.cake

```diff
@@ -0,0 +1,132 @@
+#addin nuget:?package=Cake.Git
+using System.Text.RegularExpressions;
+
+//////////////////////////////////////////////////////////////////////
+// ARGUMENTS
+//////////////////////////////////////////////////////////////////////
+
+var target = Argument("target", "Default");
+var configuration = Argument("configuration", "Release");
+
+//////////////////////////////////////////////////////////////////////
+// PREPARATION
+//////////////////////////////////////////////////////////////////////
+
+// Define directories.
+var buildDir = Directory("./src/Syncromatics.Clients.WaySine/bin") + Directory(configuration);
+var testDir = Directory("./tests/Syncromatics.Clients.WaySine.IntegrationTests/bin") + Directory(configuration);
+
+//////////////////////////////////////////////////////////////////////
+// TASKS
+//////////////////////////////////////////////////////////////////////
+var version = "";
+var semVersion = "";
+Task("GetVersion")
+    .Does(() =>
+    {
+        var repositoryPath = Directory(".");
+        var travisTag = EnvironmentVariable("TRAVIS_TAG");
+        if (!string.IsNullOrEmpty(travisTag))
+        {
+            Information($"GetVersion: Current tag is {travisTag}");
+            version = semVersion = travisTag;
+        }
+        else
+        {
+            var travisPullRequest = EnvironmentVariable("TRAVIS_PULL_REQUEST");
+            var travisBranch = EnvironmentVariable("TRAVIS_BRANCH");
+            var branch = !string.IsNullOrEmpty(travisPullRequest) && travisPullRequest != "false"
+                ? $"pr-{travisPullRequest}-{EnvironmentVariable("TRAVIS_PULL_REQUEST_BRANCH")}"
+                : !string.IsNullOrEmpty(travisBranch)
+                    ? travisBranch
+                    : GitBranchCurrent(repositoryPath).FriendlyName;
+
+            Information($"GetVersion: Current branch is {branch}");
+            var prereleaseTag = Regex.Replace(branch, @"\W+", "-");
+            var describe = GitDescribe(repositoryPath, GitDescribeStrategy.Tags);
+            var isMaster = prereleaseTag == "master" || prereleaseTag == "-no-branch-";
+            version = string.Join(".", describe.Split(new[] { '-' }, 3).Take(2));
+            semVersion = version + (isMaster ? "" : $"-{prereleaseTag}");
+        }
+        Information($"Version: {semVersion}");
+    });
+
+Task("Clean")
+    .Does(() =>
+    {
+        CleanDirectory(buildDir);
+        CleanDirectory(testDir);
+    });
+
+Task("Build")
+    .IsDependentOn("InnerTest");
+
+Task("Package")
+    .IsDependentOn("InnerPackage");
```

Comment:
What's the reason for this again?

Reply:
I copied the tasks from an existing repository.  I think it was initially spinning up its own docker container and doing the Inner* tasks in there, but that's no longer the case.  @thzinc do we still need the Inner* tasks for anything?